wordpress-mobile/WordPress-iOS
# Comment Detail: Render content in web view

Refs #17108, #17087, `aXjNk0P-zpl`

This adds the web view implementation to render the comment content. Some notes:
- Fixed a little bug when fetching the user's avatar URL.
- `richCommentTemplate.html` and `richCommentStyle.css` were added to stylize the contents according to the designs.
- Navigation requests from the web view are temporarily disabled for now. i.e. links look clickable but will do nothing (for now). It'll be implemented in the next PR.
- To handle content resize on orientation change, I ended up reloading the contents (i.e. calling `loadHTMLString` on the web view) again. 
  - I've tried other methods (`reload()`, through javascript's `location.reload()`, re-querying `scrollHeight`, ...) to no avail. The resize only works by calling `loadHTMLString` again. Somewhat expensive, I know. 😅 But it's what works so far and I've yet to find a better way.
- All resources within the comment content (images, video thumbnails) are loaded synchronously for now. For heavy comments (e.g. with >5 images), it may take a couple of seconds for the web view to fully render the content.
  - I'll take a look at loading the images asynchronously in the next PR. My assumption is it'll _at least_ cause the table to recalculate the cell height after each image completes loading, but if the method proves too tricky, I'll defer the async improvement and instead show a loading animation on top of the web view for the much simpler approach.
- Lastly, cell/component visibility based on user permission will be implemented in the upcoming PR.

## To test

- Ensure that the `New Comment Detail` flag is enabled.
- Go to My Site > Comments, and pick any comment.
- Verify that the comment content is displayed correctly (and beautifully ✨). 
  - Note that you may need to jump to multiple sites and hunt for tricky comments that may break the layout or appear strange. If you find them, please let me know! 
- Try changing to landscape orientation, and verify that the content is resized properly.

## Regression Notes
1. Potential unintended areas of impact
n/a. Feature is still hidden behind a flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a. Feature is still hidden behind a flag.

3. What automated tests I added (or what prevented me from doing so)
n/a. Feature is still hidden behind a flag.

PR submission checklist:
